packages/cli/src/common/zip.js

```diff
@@ -0,0 +1,43 @@
+const path = require("path");
+const fs = require("fs");
+const os = require("os");
+const archiver = require("archiver");
+
+/**
+ * Compress the directory contents of `directoryPath` and creates a zip archive in the os temp dir
+ * @param {string} directoryPath - Directory to compress
+ * @returns {Promise<string>} - Promise with the path of zip file created
+ */
+function zipDirectory(directoryPath) {
+  const tempDir = os.tmpdir();
+  if (!fs.existsSync(tempDir)) {
+    fs.mkdirSync(tempDir);
+  }
+  const zipPath = path.join(tempDir, "SPAShipArchive.zip");
```

Comment:
I would like to see the zip file given a dynamic name, rather than hardcoded "SPAShipArchive.zip".  I'd suggest using [read-pkg-up](https://github.com/sindresorhus/read-pkg-up) to find the nearest package.json, and using the `name` and `version` from it to build a filename.  Similar to what `npm pack` does.  The resulting filename would be something like:

```
MyApp-v1.0.1.zip
```